x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/rule_switch/index.tsx

```diff
@@ -62,13 +64,29 @@ export const RuleSwitchComponent = ({
         await enableRulesAction([id], event.target.checked!, dispatch, dispatchToaster);
       } else {
         try {
-          const updatedRules = await enableRules({
+          const enabling = event.target.checked!;
+          const response = await enableRules({
             ids: [id],
-            enabled: event.target.checked!,
+            enabled: enabling,
           });
-          setMyEnabled(updatedRules[0].enabled);
-          if (onChange != null) {
-            onChange(updatedRules[0].enabled);
+          const { rules, errors } = bucketRulesResponse(response);
+
+          if (errors.length > 0) {
+            setMyIsLoading(false);
+            const title = enabling
+              ? i18n.BATCH_ACTION_ACTIVATE_SELECTED_ERROR(1)
+              : i18n.BATCH_ACTION_DEACTIVATE_SELECTED_ERROR(1);
+            displayErrorToast(
+              title,
+              errors.map(e => e.error.message),
+              dispatchToaster
+            );
+          } else {
+            const [rule] = rules;
```

Comment:
Did you want a:

```ts
setMyIsLoading(false);
```

here?

Reply:
I see there is an extra one at the bottom so up to you on how you want to do this one. You can disregard my comment if you want.